src/cascadia/TerminalConnection/AzureConnection.cpp

```diff
@@ -370,15 +334,15 @@ namespace winrt::Microsoft::Terminal::TerminalConnection::implementation
                         }
                         catch (...)
                         {
-                            // Websocket has been closed
-                            if (!_closing.load())
+                            // Websocket has been closed; consider it a graceful exit?
+                            // This should result in our termination.
+                            if (_transitionToState(ConnectionState::Closed))
                             {
-                                _state = State::NoConnect;

```

Comment:
Should we just remove the `AzureState::NoConnect` state entirely now?

Reply:
I'm working on the AZ conn in my free time, I'll defer this to then.

Reply:
er defer this to that work